examples/serial.c

```diff
@@ -0,0 +1,42 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <unistd.h>
+#include <stdint.h>
+#include <pthread.h>
+
+#include "wiringX.h"
+
+int fd=-1;
+int date_receive=0;
+unsigned char data_send=0x00;
+
+void *interrupt(void *param) {
+	while(1){
+		if(wiringXSerialDataAvail(fd)>0) {
+			date_receive=wiringXSerialGetChar(fd);
+			printf("Data received is: %d.\n", date_receive);
+			sleep(1);
+		}
+	}
+}
+
+int main(void) {
+	pthread_t pth;
+
+	wiringXSetup();
+	pinMode(0, OUTPUT);
+	if ((fd = wiringXSerialOpen("/dev/ttyS0", 9600)) < 0) {
+	    fprintf(stderr, "Unable to open serial device: %s\n", strerror (errno)) ;
+	    return -1;
+	}
+
+	pthread_create(&pth, NULL, interrupt, NULL);
+
+	wiringXSerialPutChar(fd, data_send);
+	while(1) {
+		digitalWrite(0, HIGH);
+		sleep(1);
+		digitalWrite(0, LOW);
+		sleep(1);
+	}
```

Comment:
But I think it's necessary to show how to receive data in an interrupt, so users can do other things in main.